packages/paste-core/components/popover/__tests__/index.spec.tsx

```diff
@@ -78,4 +84,128 @@ describe('Popover', () => {
       expect(results).toHaveNoViolations();
     });
   });
+
+  describe('Customization', () => {
+    it('should set default data-paste-element attribute on Popover and customizable children', (): void => {
+      render(
+        <CustomizationProvider baseTheme="default">
+          <PopoverContainer baseId="test-id">
+            <PopoverButton variant="primary" data-testid="popover-button">
+              Open popover
+            </PopoverButton>
+            <Popover aria-label="Popover" data-testid="popover">
+              <Text as="span">This is the Twilio styled popover that you can use in all your applications.</Text>
+            </Popover>
+          </PopoverContainer>
+        </CustomizationProvider>
+      );
+
+      const popoverComp = screen.getByTestId('popover');
+      const popoverButton = screen.getByTestId('popover-button');
+
+      expect(popoverComp.querySelector('[data-paste-element="POPOVER"]')).toBeInTheDocument();
+      expect(popoverComp.querySelector('[data-paste-element="POPOVER_CLOSE_BUTTON"]')).toBeInTheDocument();
+      expect(popoverComp.querySelector('[data-paste-element="POPOVER_CLOSE_ICON"]')).toBeInTheDocument();
+
+      expect(popoverButton).toHaveAttribute('data-paste-element', 'POPOVER_BUTTON');
+    });
+
+    it('should add custom styles to Popover and customizable children', (): void => {
+      render(
+        <CustomizationProvider
+          baseTheme="default"
+          elements={{
+            POPOVER: {
+              backgroundColor: 'colorBackground',
+            },
+            POPOVER_BUTTON: {
+              backgroundColor: 'colorBackgroundSuccess',
+            },
+            POPOVER_CLOSE_BUTTON: {
+              backgroundColor: 'colorBackgroundBodyInverse',
+            },
+            POPOVER_CLOSE_ICON: {
+              color: 'colorTextInverse',
+            },
+          }}
+        >
+          <PopoverContainer baseId="test-id">
+            <PopoverButton variant="primary" data-testid="popover-button">
+              Open popover
+            </PopoverButton>
+            <Popover aria-label="Popover" data-testid="popover">
+              <Text as="span">This is the Twilio styled popover that you can use in all your applications.</Text>
+            </Popover>
+          </PopoverContainer>
+        </CustomizationProvider>
+      );
+
+      const popoverComp = screen.getByTestId('popover');
+      const popoverButton = screen.getByTestId('popover-button');
+
+      expect(popoverComp.querySelector('[data-paste-element="POPOVER"]')).toHaveStyleRule(
```

Comment:
You might be able to just merge this `it()` block with the previous one, and check for the style and data-paste-element in the same block.  Would simplify the test file by not needing the repetitive render boilerplate.